script.js

```diff
@@ -1,4 +1,140 @@
+//Base Done
+
+var currentGameMode = "waiting for name";
+var userName = "";
+var totalGamesPlayed = 0;
+var totalGamesWon = 0;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var myOutputValue = "";
+  if (currentGameMode == "waiting for name") {
+    userName = input;
+    // switch game mode to Scissors Paper Stone once name is given
+    currentGameMode = "scissors paper stone game";
+    myOutputValue =
+      "hello " +
+      input +
+      ", you are now playing scissors paper stone! Please only choose the following:<br>✂️scissors<br>🧻paper<br><b>or</b> 🪨stone";
+    return myOutputValue;
+  } else if (currentGameMode == "scissors paper stone game") {
+    // scissors paper stone logic added here
+    var computerRandom = generateRandomOutcome();
+    var myOutputValue = `${userName}, you chose ${input},<br>the computer chose ${computerRandom}`;
+    console.log("Computer Chose: " + computerRandom);
+    console.log(`User Chose: ${input}`);
+    if (
+      (input == "scissors" && computerRandom == "paper") ||
+      (input == "paper" && computerRandom == "stone") ||
+      (input == "stone" && computerRandom == "scissors")
+    ) {
+      totalGamesPlayed = totalGamesPlayed + 1;
+      totalGamesWon = totalGamesWon + 1;
+      if (totalGamesWon / totalGamesPlayed >= 0.5) {
+        return `🎉 Congratulations! <br> ${userName}, you won! 🤗<br>You chose ${input} and the computer chose ${computerRandom}! <br> You have been winning ${totalGamesWon} out of ${totalGamesPlayed} rounds! It's looking good!`;
+      } else {
+        return `🎉 Congratulations! <br> ${userName}, you won! 🤗<br>You chose ${input} and the computer chose ${computerRandom}! <br> You have been winning ${totalGamesWon} out of ${totalGamesPlayed} rounds! Try harder!`;
```

Comment:
Similarly over here, you can have a helper function called message(totalGamesWon, totalGamesPlayed) that returns the message